bundles/tools.mdsd.ecoreworkflow.mwe2lib/src/tools/mdsd/ecoreworkflow/mwe2lib/component/GapPatternPostProcessor.java

```diff
@@ -29,14 +29,12 @@
 import org.eclipse.emf.mwe.core.monitor.ProgressMonitor;
 
 public class GapPatternPostProcessor extends AbstractWorkflowComponent2 {
-	private static final String CLASSNAME_MATCHER_PATTERN =
-			"(?<=[^a-zA-Z\\d_$])(%s)(?=[^a-zA-Z\\d_$])";
-	
 	private static final String JAVAFILE_MATCHER_PATTERN = 
 			"([^\\n\\s]*).java$";
 
 	private static final Log LOG = LogFactory.getLog(GapPatternPostProcessor.class);
 	private final Collection<GapPatternFolderSet> folders = new LinkedList<>();
+	private String searchPattern = "(?<!\\bnew\\W)\\b(%s)(?=[^a-zA-Z\\d_$])(?!\\W+eINSTANCE\\b)";
```

Comment:
My regex skills might be too bad but I had to google many parts of this regex and am still not sure if it does what it is intended to be. Could you add some documentation describing the intention of the parts or maybe a complete breakdown of the whole expression, please?

Reply:
I've been testing the new regex for some local generations. As far as I can tell, it captures the relevant occurences.